teuthology/suite/run.py

```diff
@@ -345,31 +333,40 @@ def schedule_suite(self):
                 args=arg
             )
 
+            sha1 = self.base_config.sha1
             if config.suite_verify_ceph_hash:
                 full_job_config = dict()
                 deep_merge(full_job_config, self.base_config.to_dict())
                 deep_merge(full_job_config, parsed_yaml)
                 flavor = util.get_install_task_flavor(full_job_config)
-                sha1 = self.base_config.sha1
                 # Get package versions for this sha1, os_type and flavor. If
                 # we've already retrieved them in a previous loop, they'll be
                 # present in package_versions and gitbuilder will not be asked
                 # again for them.
-                package_versions = util.get_package_versions(
-                    sha1,
-                    os_type,
-                    flavor,
-                    package_versions
-                )
+                try:
+                    Run.package_versions = util.get_package_versions(
+                        sha1,
+                        os_type,
+                        flavor,
+                        Run.package_versions
+                    )
+                except VersionNotFoundError:
+                    pass
                 if not util.has_packages_for_distro(sha1, os_type, flavor,
-                                                    package_versions):
+                                                    Run.package_versions):
                     m = "Packages for os_type '{os}', flavor {flavor} and " + \
                         "ceph hash '{ver}' not found"
                     log.error(m.format(os=os_type, flavor=flavor, ver=sha1))
                     jobs_missing_packages.append(job)
+                    # optimization: one missing causes backtrack in newest mode,
```

Comment:
if one package is missing, that's enough to cause backtracking if you're in newest mode, so if we're in newest mode, stop right now rather than continuing to look for more missing packages.  It's pretty terse.  Would it scan if I added "package" after "missing"?